File: programmers/java/lev0/콜라츠 수열 만들기.java
import java.util.ArrayList;
class Solution {
    public int[] solution(int n) {
        var answer = new ArrayList<Integer>();
        answer.add(n);
        int i=0;
        while((int)answer.get(i)!=1){
            boolean b = answer.get(i) % 2 == 0 ? answer.add(answer.get(i) / 2) : answer.add(answer.get(i) * 3 + 1);
            i+=1;

        }
        return answer.stream().mapToInt(el->el).toArray();

        
    }
}

File: programmers/js/lev0/콜라츠 수열 만들기.js
function solution(n) {
  let answer = [n];
  let i = 0;
  while (answer[i] != 1) {
    answer[i] % 2 == 0
      ? answer.push(answer[i] / 2)
      : answer.push(answer[i] * 3 + 1);
    i += 1;
  }

  return answer;
}

// function solution(n, arr = []) {
//     arr.push(n)
//     if (n === 1) return arr
//     if (n % 2 === 0) return solution(n / 2, arr)
//     return solution(3 * n + 1, arr)
// }
